university-agent/src/main/java/nl/quintor/studybits/service/StudentService.java

```diff
@@ -75,21 +78,38 @@ public void setStudentDid(String studentId, String studentDid) {
 
     @Transactional
     public Student createStudent() {
-        Student student = new Student();
-        student.setStudentId(UUID.randomUUID().toString());
+        String randomId = UUID.randomUUID().toString();
+        if(!studentExists(randomId)) {
+            Student student = new Student();
+            student.setStudentId(randomId);
 
-        return studentRepository.saveAndFlush(student);
+            return studentRepository.saveAndFlush(student);
+        }
+
+        throw new EntityExistsException();
     }
 
     @Transactional
     public Student createStudent(String id, String password, String did) {
-        Student student = new Student();
-        student.setStudentId(id);
-        String passwordHash = bCryptPasswordEncoder.encode(password);
-        student.setPassword(passwordHash);
-        student.setStudentDid(did);
+        if(!studentExists(id)) {
+            Student student = new Student();
+            student.setStudentId(id);
+            String passwordHash = bCryptPasswordEncoder.encode(password);
+            student.setPassword(passwordHash);
+            student.setStudentDid(did);
+
+            return studentRepository.saveAndFlush(student);
+        }
+        //TODO: Return this exception as response instead of a 401
```

Comment:
I think this is now translated to a 409, which is fine, so this comment can be removed?

Reply:
It actually is translated to a 401 somewhere up the chain